.github/workflows/deploy.yml

```diff
@@ -0,0 +1,102 @@
+# ref: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows
+
+name: Publish Python 🐍 distribution 📦 to PyPI and TestPyPI
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+
+  # production build
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.x"
+
+    - name: Install pypa/build
+      run: >-
+        python3 -m
+        pip install
+        wheel sdist setuptools_scm
+        --user
+
+    - name: Build a binary wheel and a source tarball
+      run: python3 setup.py sdist bdist_wheel
+
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+
+
+  # This will be triggered only when a tag is pushed
+  publish-to-pypi:
+    name: Publish Python 🐍 distribution 📦 to PyPI
+    if: startsWith(github.ref, 'refs/tags/')
+    needs:
+    - build
+    runs-on: ubuntu-latest
+
+    environment:
+      name: pypi
+      url: https://pypi.org/p/stock-indicators
+
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+
+    steps:
+
+    - name: Download all the dists
+      uses: actions/download-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+
+    - name: Publish distribution 📦 to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+
+    - name: Tag and draft release note
```

Comment:
Maybe I was wrong here.  I think you can create a Tag when you _manually_ create a [draft] Release record, then wait for it to deploy, then make the Release a non-Draft?  The only problem with that approach is that it wouldn't auto-generate the release notes.

Reply:
(other than bypassing branch protection on `main` and pushing a Tag directly)

Reply:
Oh, wait, you can manually add [automatically generated release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)

Reply:
> I really only have one question, how is the release tag introduced on main?

To answer my own question, you can manually create a Draft Release, add a new custom Tag with the version number, add auto-generated release notes; then Save; wait for deployment, remove Draft status from Release note.  Is that the approach here, or did you have something else in mind?

Reply:
Ah, I was planning to push tag from my local repository.

Reply:
Feature branch tags get squashed (removed) when you merge to `main`, in our normal merge strategy.  We'd have to push a solitary tag directly to `main` if we want to keep Squash merge limitations (which is nicer commit history).

Reply:
You mean I cannot push a tag directly, because I'm working on forked repository. hmm okay.
Let's just remove `Tag and draft release note` step and use GH native feature!

Reply:
Go for it, let's give is a try.  Just a couple other removals for cleanup below too.